src/linux/mod.rs

```diff
@@ -165,11 +182,46 @@ impl<'a> SharedLibraryTrait for SharedLibrary<'a> {
 
     #[inline]
     fn name(&self) -> &CStr {
-        self.name
+        &*self.name
     }
 
-    #[inline]
     fn id(&self) -> Option<SharedLibraryId> {
+        fn align(alignment: usize, offset: &mut usize) {
+            let diff = *offset % alignment;
+            if diff != 0 {
+                *offset += alignment - diff;
+            }
+        }
+
+        unsafe {
+            for segment in self.segments() {
+                if segment.phdr.as_ref().unwrap().p_type != libc::PT_NOTE {
+                    continue;
+                }
+
+                let mut offset = (*segment.phdr).p_offset as usize;
+                let mut alignment = (*segment.phdr).p_align as usize;
+                if alignment < 4 {
+                    alignment = 4;
+                }
```

Comment:
This should never happen, but I think technically we should check that alignment is 4 or 8.

Reply:
Apparently that's what readelf does. I took it from this commit in goblin: https://github.com/m4b/goblin/commit/467d7d70a10fe908030e9d18f69241d8f11a5bec

Reply:
Sure, but you're not checking the resulting alignment is [`4|8`](https://github.com/m4b/goblin/commit/467d7d70a10fe908030e9d18f69241d8f11a5bec#diff-9a4d0183cba0574248ca79ac15654fb4R178).

Reply:
Oh you are right, i accidentally removed this.